cmd/av/stack_sync.go

```diff
@@ -77,71 +80,172 @@ stack. This is useful for rebasing a whole stack on the latest changes from the
 		if err != nil {
 			return err
 		}
-		if conflict {
+		var branchesToSync []string
+		if state.Continue {
+			// If we're continuing, we assume the previous branches are already
+			// synced correctly and we just need to sync the subsequent
+			// branches. (This matters because if we're here, that means there
+			// was a sync conflict, and we need to `git rebase --continue`
+			// before we can sync the next branch, and git will scream at us if
+			// we try to do something in the repo before we finish that)
+			branchesToSync = []string{state.CurrentBranch}
+		} else {
+			// Otherwise, this is not a --continue, so we want to sync every
+			// ancestor first.
 			currentBranch, err := repo.CurrentBranchName()
 			if err != nil {
 				return err
 			}
-			if err := writeStackSyncState(repo, &stackSyncState{
-				CurrentBranch: currentBranch,
-				Config:        config,
-			}); err != nil {
+			branchesToSync, err = previousBranches(branches, currentBranch)
+			if err != nil {
+				return err
+			}
+			branchesToSync = append(branchesToSync, currentBranch)
+		}
+		// Either way (--continue or not), we sync all subsequent branches.
+		nextBranches, err := subsequentBranches(branches, branchesToSync[len(branchesToSync)-1])
+		if err != nil {
+			return err
+		}
+		branchesToSync = append(branchesToSync, nextBranches...)
+
+		logrus.WithField("branches", branchesToSync).Debug("determined branches to sync")
+		conflict := false
+	loop:
+		for _, currentBranch := range branchesToSync {
+			state.CurrentBranch = currentBranch
+			currentMeta, ok := branches[currentBranch]
+			if !ok {
+				return errors.Errorf("stack metadata not found for branch %q", currentBranch)
+			}
+			if currentMeta.Parent == "" {
+				continue
+			}
+			log := logrus.WithFields(logrus.Fields{
+				"current": currentBranch,
+				"parent":  currentMeta.Parent,
+			})
+
+			// Checkout the branch (unless we need to continue a rebase, in which
+			// case Git will yell at us)
+			var res *stacks.SyncResult
+			if state.Continue {
+				log.Debug("finishing previous interrupted sync...")
+				res, err = stacks.SyncContinue(repo, stacks.StrategyRebase)
+
+				// Only the first branch needs to be --continue'd, for the rest
+				// we just do a normal merge/rebase
+				state.Continue = false
+			} else {
+				log.Debug("syncing branch...")
+				_, err := repo.CheckoutBranch(&git.CheckoutBranch{Name: currentBranch})
+				if err != nil {
+					return err
+				}
+				res, err = stacks.SyncBranch(repo, &stacks.SyncBranchOpts{
+					Parent:   currentMeta.Parent,
+					Strategy: stacks.StrategyRebase,
+				})
+			}
+			log.WithField("result", res).WithError(err).Debug("sync finished")
+			if err != nil {
+				return errors.WrapIff(err, "failed to sync branch %q", currentBranch)
+			}
+
+			fmt.Printf("Syncing %q into %q: ", currentMeta.Parent, currentBranch)
+			switch res.Status {
+			case stacks.SyncAlreadyUpToDate:
+				fmt.Println("already up-to-date (no-op)")
```

Comment:
do we need these print statements? Feel like `log.Debug`. If these should be print, then maybe descrbe more:
`updated branch x from y.`
`conflict found while syncing x from y, please resolve and call av stack sync --continue`

Reply:
Not sure if you missed the print statement above, but ultimately we get:

```
$ av stack sync
Syncing "xyz-1" into "xyz-2": already up-to-date (no-op)
Syncing "xyz-2" into "xyz-3": conflict
error: could not apply 29da82e... 3a
    Resolve all conflicts manually, mark them as resolved with
    "git add/rm <conflicted_files>", then run "git rebase --continue".
    You can instead skip this commit: run "git rebase --skip".
    To abort and get back to the state before "git rebase", run "git rebase --abort".
    Could not apply 29da82e... 3a

error: conflict detected: please resolve and then run `av stack sync --continue`
```